x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/actions/duplicate_exceptions.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ExceptionListClient } from '@kbn/lists-plugin/server';
+import type { ListArray } from '@kbn/securitysolution-io-ts-list-types';
+import { ExceptionListTypeEnum } from '@kbn/securitysolution-io-ts-list-types';
+
+import type { RuleParams } from '../../../rule_schema';
+
+export const duplicateExceptions = async (
+  ruleId: RuleParams['ruleId'],
+  exceptionLists: RuleParams['exceptionsList'],
+  shouldDuplicate: boolean,
+  exceptionsClient: ExceptionListClient | undefined
+): Promise<RuleParams['exceptionsList']> => {
+  if (exceptionLists == null) {
+    return [];
+  }
+
+  // Sort the shared lists and the rule_default lists.
+  // Only a single rule_default list should exist per rule.
+  const [[ruleDefaultList], sharedLists] = exceptionLists.reduce<[ListArray, ListArray]>(
+    (acc, list) => {
+      const [def, shared] = acc;
+      if (list.type === ExceptionListTypeEnum.RULE_DEFAULT) {
+        return [[list], shared];
+      } else {
+        return [def, [...shared, list]];
+      }
+    },
+    [[], []]
+  );
+  // If user does not want exceptions duplicated, return empty array.
+  // This will remove the shared list references between rule<-->exceptions.
+  if (!shouldDuplicate) {
+    return [];
+  }
```

Comment:
Can we move this logic outside this function and skip calling completely if `shouldDuplicate` is false?